packages/template-retail-react-app/scripts/translations/extract-default-messages.js

```diff
@@ -16,77 +16,68 @@ const fs = require('fs')
+        const command = [
+            'formatjs extract "app/**/*.{js,jsx,ts,tsx}"',
+            `--out-file translations/${locale}.json`,
+            '--id-interpolation-pattern [sha512:contenthash:base64:6]'
+        ].join(' ')
         exec(command, (err) => {
             if (err) {
                 console.error(err)
             }
         })
     } else {
-        const overridesPath = path.join(process.cwd(), pkgJSON.ccExtensibility?.overridesDir)
+        const overridesPath = path.join(process.cwd(), overridesDir)
         // get all the files in extended app
         const files = getAllFilesByExtensions(
             path.join(overridesPath, 'app'),
             [],
-            ['js', 'jsx', 'ts', 'tsx']
+            ['.js', '.jsx', '.ts', '.tsx']
         )
         // get the file names that are overridden in base template
         const overriddenFiles = files
-            .map((path) => {
-                const replacedPath = path.replace(
-                    overridesDir,
-                    `node_modules/${pkgJSON.ccExtensibility?.extends}`
-                )
-                // check if this file does exist in base template
-                const isFileExist = fs.existsSync(replacedPath)
-                return isFileExist ? replacedPath : ''
-            })
-            .filter(Boolean)
-        // rename the files needs to ignore to have .ignore extensions so it won't be recognized by formatjs
-        overriddenFiles.forEach((filePath) => {
-            fs.rename(filePath, `${filePath}.ignore`, (err) => err && console.error(err))
-        })
-
-        const extractCommand = `formatjs extract "./node_modules/${pkgJSON.ccExtensibility?.extends}/app/**/*.{js,jsx,ts,tsx}" "${pkgJSON.ccExtensibility?.overridesDir}/app/**/*.{js,jsx,ts,tsx}" --out-file translations/${locale}.json --id-interpolation-pattern [sha512:contenthash:base64:6]`
+            .map((path) => path.replace(overridesDir, `node_modules/${extendsPkg}`))
+            .filter((file) => fs.existsSync(file))
+        const extractCommand = [
+            'formatjs extract',
+            '"./node_modules/${extendsPkg}/app/**/*.{js,jsx,ts,tsx}"',
+            '"${overridesDir}/app/**/*.{js,jsx,ts,tsx}"',
+            `--out-file translations/${locale}.json`,
+            '--id-interpolation-pattern [sha512:contenthash:base64:6]',
+            '--ignore',
+            ...overriddenFiles.map((file) => `'${file}'`)
+        ].join(' ')
         exec(extractCommand, (err) => {
             if (err) {
                 console.error(err)
             }
-            // restore file names
-            overriddenFiles.forEach((filePath) => {
-                fs.rename(`${filePath}.ignore`, filePath, (err) => err && console.error(err))
-            })
         })
     }
+}
+
+try {
+    // example usage: node ./scripts/translations/extract-default-messages en-US en-GB
+    process.argv.slice(2).forEach(extract)
```

Comment:
Nice! 👍

Reply:
@wjhsf is this opt-in? E.g. can a customer opt out of the behavior if it's not desired? I see in the package.json change that we are explicit about `<command> en-US en-GB` so that implies "yes" to me

Reply:
The customer can specify whatever locales they choose, if that's what you're asking about.